chain-impl-mockchain/src/ledger/evm.rs

```diff
@@ -31,19 +86,59 @@ impl EvmState for super::Ledger {
         &self.evm.environment
     }
 
-    fn account(&self, address: Address) -> Option<Account> {
-        self.evm.accounts.get(&address).cloned()
+    fn account(&self, evm_address: EvmAddress) -> Option<EvmAccount> {
+        match self.evm.adress_mapping.jor_address(evm_address) {
+            Some(jor_address) => match self.accounts.get_state(&jor_address) {
+                Ok(account) => Some(EvmAccount {
+                    balance: account.value.0.into(),
+                    state: account.evm_state.clone(),
+                }),
+                Err(_) => None,
+            },
+            None => None,
+        }
     }
 
-    fn contains(&self, address: Address) -> bool {
-        self.evm.accounts.contains(&address)
+    fn contains(&self, address: EvmAddress) -> bool {
+        self.evm.adress_mapping.jor_address(address).is_some()
     }
 
-    fn modify_account<F>(&mut self, address: Address, f: F)
+    fn modify_account<F>(&mut self, address: EvmAddress, f: F) -> Result<(), String>
     where
-        F: FnOnce(Account) -> Option<Account>,
+        F: FnOnce(EvmAccount) -> Option<EvmAccount>,
     {
-        self.evm.accounts = self.evm.accounts.clone().modify_account(address, f);
+        let address = self.evm.adress_mapping.jor_address(address).ok_or(format!(
+            "Can not find corresponding jormungadr account for the evm account: {}",
+            address
+        ))?;
+
+        let account = self
+            .accounts
+            .get_state(&address)
+            .cloned()
+            .unwrap_or_else(|_| JorAccount::new(Value::zero(), ()));
+
+        match f(EvmAccount {
+            balance: account.value.0.into(),
+            state: account.evm_state,
+        }) {
+            Some(account) => {
+                self.accounts = self
+                    .accounts
+                    .evm_update(
```

Comment:
Yes, you are right, it is not used in this PR, but such option will be used later with the evm contract mapping implementation.

Reply:
Maybe it could be useful to add two different methods so that invocations like this one are cleaner?

Reply:
fixed

Reply:
>Yes, you are right, it is not used in this PR, but such option will be used later with the evm contract mapping implementation.

Sorry, was mistaken and confused you. We probably can create a new account because of 
```
        let account = self
            .accounts
            .get_state(&address)
            .cloned()
            .unwrap_or_else(|_| JorAccount::new(Value::zero(), ()));
```